.github/workflows/buiild-and-stage-backend-function.yml

```diff
@@ -0,0 +1,51 @@
+# This workflow will build a backend cloud function with Gradle
+# For more information see: https://github.com/google-github-actions/deploy-cloud-functions
+
+name: Build and stage backend cloud function
+
+on:
+  push:
+    branches: [ master ]
+
+jobs:
+  build-and-stage:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up AdoptOpenJDK 11
+        uses: joschi/setup-jdk@v2
+        with:
+          java-version: 11
+
+      - name: Cache Gradle packages
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: v1-${{ runner.os }}-gradle-${{ hashfiles('**/gradle-wrapper.properties') }}-${{ hashFiles('**/*.gradle') }}
+          restore-keys: v1-${{ runner.os }}-gradle-${{ hashfiles('**/gradle-wrapper.properties') }}
+
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+
+      # Set up ADC for credentials
+      # Ideally this step should be part of action.yml but
+      # GHA does not yet support action within action so we're stuck.
+      # Please refer to https://github.com/actions/runner/pull/612 for updates.
+      - uses: google-github-actions/setup-gcloud@master
+        with:
+          service_account_email: ${{ secrets.DSP_CF_STAGE_SA_EMAIL }}
+          service_account_key: ${{ secrets.DSP_CF_STAGE_SA_KEY }}
+          export_default_credentials: true
+
+      # Call the uild-and-stage-backend-function action
+      # inputs: archive_bucket represents the Google Bucket for staging the Cloud Function archive
+      - name: Build and stage the backend cloud function
+        id: build-and-stage
+        uses: ./.github/actions/build-and-stage-backend-function
+        with:
+          archive_bucket: cloudfunctions-dsp-artifact-registry
```

Comment:
the `build-and-stage-backend-function` action only has two steps … and those steps have prerequisites (like setting up gcloud) that only exist in this action. It doesn't feel necessary to have a separate `build-and-stage-backend-function` action; just copy its two steps into this action.